### PR TITLE
RFC-0017-Autograd-subgraphs-PT-ONNX-Export

### PR DESCRIPTION
**Pull request by shubhambhokare1**
[Rendered](https://github.com/pytorch/rfcs/blob/880aa3464d14ef29471e4a3c05f06d0c1cb43b02/RFC-0017-Autograd-subgraphs-PT-ONNX-Export.md)

### CODE REVIEW

**Comment by albanD** [comment]
Are you supposed to be able to do training on the exported model?

File: RFC-0017-Autograd-subgraphs-PT-ONNX-Export.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    bool is_inplace,
+    bool unpack_output) {
+  if (!jit::tracer::isTracing()) {
+    return;
+  }
+  // ...
+
+  for (const auto i : c10::irange(num_outputs)) {
+    PyObject* obj = PyTuple_GET_ITEM(output_objects, i);
+    if (THPVariable_Check(obj) && !getTracingState()->expand_autograd_op) {
```
Comment (BowenBao) [inline RFC-0017-Autograd-subgraphs-PT-ONNX-Export.md:107]: `&& !getTracingState()->expand_autograd_op` This is the only place of change in this code section right?

**Comment by shubhambhokare1** [comment]
@albanD yes, the intention is to be able to training on the exported model. However this change is mainly for terms of inference and exporting models with autograd functions that have complex structures that prevent us from writing symbolics for them

**Comment by albanD** [comment]
That's interesting, because removing the custom Function will change the behavior of the backward pass. So we should forbid training if we remove the custom Function right?

**Pull request closed without merging**